service/graph_utils.py

```diff
@@ -0,0 +1,74 @@
+import itertools
+
+import networkx as nx
+from networkx.drawing.nx_agraph import to_agraph
+
+
+class GraphUtils:
+
+    def handle_graph(self, vms_data, fw_rules_data):
+        nodes = [vm.vm_id for vm in vms_data]
+        edges = self.prepare_graph_edges(vms_data, fw_rules_data)
+        vms_graph = self.create_graph(nodes=nodes, edges=edges)
+        self.create_graph_visualization(vms_graph)
+        prepare_vm_potentials_attackers_dict = self.prepare_vm_potentials_attackers_dict(vms_graph)
+        return prepare_vm_potentials_attackers_dict
+
+    def prepare_graph_edges(self, vms_data, fw_rules_data):
+        tag_vms_dict = self.prepare_tag_vms_dict(vms_data)
+        edges = []
+        for fw_rule in fw_rules_data:
+            if (tag_vms_dict.get(fw_rule.source_tag) is not None) \
+                    and (tag_vms_dict.get(fw_rule.dest_tag)):
+                source_tag_vms = tag_vms_dict[fw_rule.source_tag]
+                dest_tag_vms = tag_vms_dict[fw_rule.dest_tag]
+                pairs_list = list(set(itertools.product(source_tag_vms, dest_tag_vms)))
```

Comment:
list(set())? are you expecting to have duplicate values? why not check this before?